config/src/test/kotlin/org/springframework/security/config/web/servlet/OAuth2LoginDslTests.kt

```diff
@@ -113,6 +124,74 @@ class OAuth2LoginDslTests {
         }
     }
 
+    @Test
+    fun `oauth2Login when custom authentication details source then used`() {
+        this.spring
+            .register(CustomAuthenticationDetailsSourceConfig::class.java, ClientConfig::class.java)
+            .autowire()
+        mockkObject(CustomAuthenticationDetailsSourceConfig.AUTHENTICATION_DETAILS_SOURCE)
+        every {
+            CustomAuthenticationDetailsSourceConfig.AUTHENTICATION_DETAILS_SOURCE.buildDetails(any())
+        } returns Any()
+        mockkObject(CustomAuthenticationDetailsSourceConfig.AUTHORIZATION_REQUEST_REPOSITORY)
+        every {
+            CustomAuthenticationDetailsSourceConfig.AUTHORIZATION_REQUEST_REPOSITORY.removeAuthorizationRequest(any(), any())
+        } returns OAuth2AuthorizationRequest.authorizationCode()
+            .authorizationUri("/")
+            .clientId("clientId")
+            .redirectUri("/")
+            .attributes { attributes -> attributes[OAuth2ParameterNames.REGISTRATION_ID] = "google" }
+            .build()
+
+        this.mockMvc.post("/login/oauth2/code/google") {
+            param(OAuth2ParameterNames.CODE, "code")
+            param(OAuth2ParameterNames.STATE, "state")
+        }
+        .andExpect {
+            status { is3xxRedirection() }
+        }
+
+        verify(exactly = 1) { CustomAuthenticationDetailsSourceConfig.AUTHENTICATION_DETAILS_SOURCE.buildDetails(any()) }
+    }
+
+    @EnableWebSecurity
+    open class CustomAuthenticationDetailsSourceConfig : WebSecurityConfigurerAdapter() {
+
+        companion object {
+            val AUTHENTICATION_DETAILS_SOURCE: AuthenticationDetailsSource<HttpServletRequest, *> =
+                AuthenticationDetailsSource<HttpServletRequest, Any> { Any() }
+            val AUTHORIZATION_REQUEST_REPOSITORY = object: AuthorizationRequestRepository<OAuth2AuthorizationRequest> {
+                override fun loadAuthorizationRequest(request: HttpServletRequest?): OAuth2AuthorizationRequest {
+                    throw IllegalStateException()
+                }
+
+                override fun saveAuthorizationRequest(
+                    authorizationRequest: OAuth2AuthorizationRequest?,
+                    request: HttpServletRequest?,
+                    response: HttpServletResponse?
+                ) {
+                    throw IllegalStateException()
+                }
+
+                override fun removeAuthorizationRequest(request: HttpServletRequest?): OAuth2AuthorizationRequest {
+                    throw IllegalStateException()
+                }
+            }
+        }
+
+        override fun configure(http: HttpSecurity) {
+            http {
+                oauth2Login {
+                    authenticationDetailsSource = AUTHENTICATION_DETAILS_SOURCE
+                    authorizationEndpoint {
+                        authorizationRequestRepository = AUTHORIZATION_REQUEST_REPOSITORY
+                    }
+                }
+                csrf { disable() }
```

Comment:
Instead of disabling CSRF protection, you can include the token in the request using `with(csrf())`
```
this.mockMvc.post("/login/oauth2/code/google") {
            param(OAuth2ParameterNames.CODE, "code")
            param(OAuth2ParameterNames.STATE, "state")
            with(csrf())
}
```

Reply:
Done!

I actually have an unrelated question -- when I run `./gradlew clean build integrationTest --continue --scan` locally, exactly as I see in the `pr-build-workflow.yml`, a number of `checkFormat.*` tasks fail (from modules I haven't touched), and the overall build result is a failure. Meanwhile, the PR Build check succeeds with the same source.

Any idea if the checkFormat criteria be influenced by some config outside of the repository, or maybe in the git-ignored files locally?

Reply:
@nmck257 `checkFormat` will run on files even if they are git-ignored. You can run `git clean -dfx` locally to remove any files/folders that are not in source control, and then try building again.